src/nvim/eval/typval_encode.c.h

```diff
@@ -315,309 +324,318 @@ static int _TYPVAL_ENCODE_CONVERT_ONE_VALUE(
-            assert(saved_copyID != copyID && saved_copyID != copyID - 1);
-            _mp_push(*mpstack, ((MPConvStackVal) {
-              .tv = tv,
-              .type = kMPConvPairs,
-              .saved_copyID = saved_copyID,
-              .data = {
-                .l = {
-                  .list = val_list,
-                  .li = val_list->lv_first,
-                },
-              },
-            }));
-            break;
-          }
-          case kMPExt: {
-            const list_T *val_list;
-            varnumber_T type;
-            if (val_di->di_tv.v_type != VAR_LIST
-                || (val_list = val_di->di_tv.vval.v_list) == NULL
-                || val_list->lv_len != 2
-                || (val_list->lv_first->li_tv.v_type != VAR_NUMBER)
-                || (type = val_list->lv_first->li_tv.vval.v_number) > INT8_MAX
-                || type < INT8_MIN
-                || (val_list->lv_last->li_tv.v_type != VAR_LIST)) {
-              goto _convert_one_value_regular_dict;
-            }
-            size_t len;
-            char *buf;
-            if (!encode_vim_list_to_buf(val_list->lv_last->li_tv.vval.v_list,
-                                        &len, &buf)) {
-              goto _convert_one_value_regular_dict;
-            }
-            TYPVAL_ENCODE_CONV_EXT_STRING(tv, buf, len, type);
-            xfree(buf);
-            break;
-          }
-        }
-        break;
-      }
-_convert_one_value_regular_dict: {}
-      const int saved_copyID = tv->vval.v_dict->dv_copyID;
-      _TYPVAL_ENCODE_DO_CHECK_SELF_REFERENCE(tv->vval.v_dict, dv_copyID, copyID,
-                                             kMPConvDict);
-      TYPVAL_ENCODE_CONV_DICT_START(tv, tv->vval.v_dict,
-                                    tv->vval.v_dict->dv_hashtab.ht_used);
-      assert(saved_copyID != copyID && saved_copyID != copyID - 1);
-      _mp_push(*mpstack, ((MPConvStackVal) {
-        .tv = tv,
-        .type = kMPConvDict,
-        .saved_copyID = saved_copyID,
-        .data = {
-          .d = {
-            .dict = tv->vval.v_dict,
-            .dictp = &tv->vval.v_dict,
-            .hi = tv->vval.v_dict->dv_hashtab.ht_array,
-            .todo = tv->vval.v_dict->dv_hashtab.ht_used,
-          },
-        },
-      }));
-      TYPVAL_ENCODE_CONV_REAL_DICT_AFTER_START(tv, tv->vval.v_dict,
-                                               _mp_last(*mpstack));
-      break;
-    }
-    case VAR_UNKNOWN: {
-      EMSG2(_(e_intern2), STR(_TYPVAL_ENCODE_CONVERT_ONE_VALUE) "()");
-      return FAIL;
-    }
-  }
-typval_encode_stop_converting_one_item:
-  return OK;
-  // Prevent “unused label” warnings.
-  goto typval_encode_stop_converting_one_item;
-}
-
-TYPVAL_ENCODE_SCOPE int _TYPVAL_ENCODE_ENCODE(
-    TYPVAL_ENCODE_FIRST_ARG_TYPE TYPVAL_ENCODE_FIRST_ARG_NAME,
-    typval_T *const tv, const char *const objname)
-  REAL_FATTR_NONNULL_ARG(2, 3) REAL_FATTR_WARN_UNUSED_RESULT;
+#define DEFINE_TYPVAL_ENCODE_CONVERT_ONE_VALUE( \
```

Comment:
This may not be accepted. The key point of the `.c.h` file is that ***I can debug errors in these functions***. For this they *must not* be defined by a macros.